File: By-Swift/By-Swift/Programmers/월간코드챌린지/약수의개수와덧셈.swift
//
//  약수의개수와덧셈.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/05/15.
// 
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var answer = 0
    
    for number in left...right{
        if floor(sqrt(Double(number))) == sqrt(Double(number)) {
            answer -= number
        } else {
            answer += number
        }
    }
    return answer
}

File: By-Python/programmers/[월간코드챌린지]/약수의개수와덧셈.py
# 약수의 개수와 덧셈

def solution(left, right):
    answer = 0   
    
    for number in range(left,right+1):
        divisor = set()
        divisor.add(1)
        for j in range(1, (number//2)+1):
            if number % j == 0:
                divisor.add(j)
                if j != (number // j):
                    divisor.add(number//j)  
        if len(divisor) % 2 == 0:
            answer += number
        else:
            answer -= number
            
    return answer